### PR TITLE
feat(help): Example section improvements

### PR DESCRIPTION
**Pull request by jqnatividad**

### CODE REVIEW

File: src/help_markdown_gen.rs [hunk 1]
```diff
@@ -1112,16 +1112,47 @@ fn format_examples(lines: &[String]) -> String {
...
                 in_code_block = false;
             }
             let comment = trimmed.trim_start_matches('#').trim();
-            let _ = write!(md, "> {}\n\n", linkify_bare_urls(comment));
+            let _ = write!(md, "> {}\n", linkify_bare_urls(comment));
+            // Check if the next non-empty line is also a comment — if not, end the blockquote
+            let next_is_comment = lines[idx + 1..]
+                .iter()
+                .find(|l| !l.trim().is_empty())
+                .is_some_and(|l| l.trim().starts_with('#'));
```
[the remark this answers is not in the document]
Comment (jqnatividad) [reply src/help_markdown_gen.rs]: Good catch on the consistency point. While `lines[idx + 1..]` is technically safe here (slicing at `len()` yields an empty slice, not a panic, since `idx` comes from `enumerate()` on the same slice), you're right that using `.get()` is more consistent with the existing `next_is_underline` check at line 1035. Switched to `lines.get(idx + 1..).and_then(...)` pattern. Also added a unit test (`test_format_examples_comment_as_last_line`) that exercises this exact edge case.

File: src/help_markdown_gen.rs [hunk 2]
```diff
@@ -1112,16 +1112,47 @@ fn format_examples(lines: &[String]) -> String {
...
+                .iter()
+                .map(|l| l.trim())
+                .find(|nt| !nt.ends_with('\\'))
+                .is_some_and(|nt| {
+                    nt.starts_with("qsv ")
+                        || nt.starts_with("$ qsv")
+                        || nt.contains("| qsv ")
+                        || nt.contains("|qsv ")
+                        || (nt.starts_with("$ ") && nt.contains("qsv "))
+                });
```
[the remark this answers is not in the document]
Comment (jqnatividad) [reply src/help_markdown_gen.rs]: Same as above — switched to `lines.get(idx + 1..).and_then(...)` for consistency. Added `test_format_examples_backslash_as_last_line` to verify a dangling backslash as the last line doesn't panic and doesn't open a code block.

File: src/help_markdown_gen.rs [hunk 3]
```diff
@@ -1112,16 +1112,47 @@ fn format_examples(lines: &[String]) -> String {
...
+            if reaches_qsv {
+                md.push_str("```console\n");
+                in_code_block = true;
+                md.push_str(trimmed);
+                md.push('\n');
+                continue;
+            }
+        }
```
[the remark this answers is not in the document]
Comment (jqnatividad) [reply src/help_markdown_gen.rs:1158]: Agreed — added 5 unit tests covering all the edge cases you listed:
- `test_format_examples_comment_as_last_line` — comment as last line (bounds check)
- `test_format_examples_backslash_as_last_line` — backslash as last line (bounds check)
- `test_format_examples_consecutive_comments` — multiple consecutive comments merged into one blockquote
- `test_format_examples_env_var_continuation_reaches_qsv` — backslash continuation leading to qsv command
- `test_format_examples_continuation_not_reaching_qsv` — backslash continuation NOT leading to qsv command
